### PR TITLE
pathlib.Path instead of os.path

### PR DESCRIPTION
**Pull request by rietesh**
As discussed in this issue [#3061](https://github.com/pyg-team/pytorch_geometric/issues/3061#issue-985437172) This PR implements the pathlib.Path keeping in mind the changes to pathlib library in python 3.8. The changes should support python 3.6 and python 3.9 for both torch 1.8.* and 1.9.*. The test and benchmark modules still use os.path which can be changed if required.

### CODE REVIEW

**Comment by rusty1s** [comment]
Thanks a lot. I think a lot of files were re-formatted which makes reviewing this PR quite complicated. Is there any possibility to revert the format changes?

**Comment by rietesh** [comment]
Thank you for the comment, the contribution guidelines had outlined the formatting requirements nevertheless I can revert the formatting commits I will push the changes as soon as possible.

**Comment by rietesh** [comment]
I have reverted the Code format changes I hope it is now easier to review changes

**Comment by rusty1s** [comment]
Thank you!

File: examples/cora.py [hunk 1]
```diff
@@ -11,7 +11,7 @@
     T.RandomNodeSplit(num_val=500, num_test=500),
     T.TargetIndegree(),
 ])
-path = osp.join(osp.dirname(osp.realpath(__file__)), '..', 'data', dataset)
+path = Path.joinpath(Path(__file__).resolve().parent, '..', 'data', dataset)
```
Comment (cthoyt) [inline examples/cora.py:14]: why not replace the `..` with a second parent?

File: examples/dgcnn_classification.py [hunk 2]
```diff
@@ -10,7 +10,7 @@
 
 from pointnet2_classification import MLP
 
-path = osp.join(osp.dirname(osp.realpath(__file__)), '..', 'data/ModelNet10')
+path = Path.joinpath(Path(__file__).resolve().parent, '..', 'data/ModelNet10')
```
Comment (cthoyt) [inline examples/dgcnn_classification.py:13]: why not split ` 'data/ModelNet10'`

File: examples/cora.py (hunk 1 above)
Comment (rietesh) [reply examples/cora.py:14]: Yes its possible `path = Path.joinpath(Path(__file__).resolve().parent.parent, 'data', dataset)` would give the same result

File: examples/dgcnn_classification.py (hunk 2 above)
Comment (rietesh) [reply examples/dgcnn_classification.py:13]: Yes that would be easy to read

**Comment by sumanthratna** [comment]
appreciate this PR @rietesh! Just wanted to bump this and see if there were any issues finalizing this PR?

**Comment by rusty1s** [comment]
Yeah, this PR got a little bit under the radar. We would need to resolve conversations and conflicts, and ensure that CI runs through.

**Comment by rietesh** [comment]
Raised a new PR https://github.com/pyg-team/pytorch_geometric/pull/4220

**Pull request closed without merging**